// @agm/core/core.module.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.coreDirectives = coreDirectives;
exports.AgmCoreModule = void 0;

var _core = require("@angular/core");

var _map = require("./directives/map");

var _circle = require("./directives/circle");

var _rectangle = require("./directives/rectangle");

var _infoWindow = require("./directives/info-window");

var _marker = require("./directives/marker");

var _polygon = require("./directives/polygon");

var _polyline = require("./directives/polyline");

var _polylinePoint = require("./directives/polyline-point");

var _kmlLayer = require("./directives/kml-layer");

var _dataLayer = require("./directives/data-layer");

var _lazyMapsApiLoader = require("./services/maps-api-loader/lazy-maps-api-loader");

var _mapsApiLoader = require("./services/maps-api-loader/maps-api-loader");

var _browserGlobals = require("./utils/browser-globals");

var _fitBounds = require("./directives/fit-bounds");

function _typeof(obj) { "@babel/helpers - typeof"; if (typeof Symbol === "function" && typeof Symbol.iterator === "symbol") { _typeof = function _typeof(obj) { return typeof obj; }; } else { _typeof = function _typeof(obj) { return obj && typeof Symbol === "function" && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }; } return _typeof(obj); }

var __decorate = void 0 && (void 0).__decorate || function (decorators, target, key, desc) {
  var c = arguments.length,
      r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc,
      d;
  if ((typeof Reflect === "undefined" ? "undefined" : _typeof(Reflect)) === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);else for (var i = decorators.length - 1; i >= 0; i--) {
    if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
  }
  return c > 3 && r && Object.defineProperty(target, key, r), r;
};

/**
 * @internal
 */
function coreDirectives() {
  return [_map.AgmMap, _marker.AgmMarker, _infoWindow.AgmInfoWindow, _circle.AgmCircle, _rectangle.AgmRectangle, _polygon.AgmPolygon, _polyline.AgmPolyline, _polylinePoint.AgmPolylinePoint, _kmlLayer.AgmKmlLayer, _dataLayer.AgmDataLayer, _fitBounds.AgmFitBounds];
}
/**
 * The angular-google-maps core module. Contains all Directives/Services/Pipes
 * of the core module. Please use `AgmCoreModule.forRoot()` in your app module.
 */


var AgmCoreModule =
/** @class */
function () {
  function AgmCoreModule() {}

  AgmCoreModule_1 = AgmCoreModule;
  /**
   * Please use this method when you register the module at the root level.
   */

  AgmCoreModule.forRoot = function (lazyMapsAPILoaderConfig) {
    return {
      ngModule: AgmCoreModule_1,
      providers: _browserGlobals.BROWSER_GLOBALS_PROVIDERS.concat([{
        provide: _mapsApiLoader.MapsAPILoader,
        useClass: _lazyMapsApiLoader.LazyMapsAPILoader
      }, {
        provide: _lazyMapsApiLoader.LAZY_MAPS_API_CONFIG,
        useValue: lazyMapsAPILoaderConfig
      }])
    };
  };

  var AgmCoreModule_1;
  AgmCoreModule = AgmCoreModule_1 = __decorate([(0, _core.NgModule)({
    declarations: coreDirectives(),
    exports: coreDirectives()
  })], AgmCoreModule);
  return AgmCoreModule;
}();

exports.AgmCoreModule = AgmCoreModule;
