// @agm/core/directives.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "AgmMap", {
  enumerable: true,
  get: function get() {
    return _map.AgmMap;
  }
});
Object.defineProperty(exports, "AgmCircle", {
  enumerable: true,
  get: function get() {
    return _circle.AgmCircle;
  }
});
Object.defineProperty(exports, "AgmRectangle", {
  enumerable: true,
  get: function get() {
    return _rectangle.AgmRectangle;
  }
});
Object.defineProperty(exports, "AgmInfoWindow", {
  enumerable: true,
  get: function get() {
    return _infoWindow.AgmInfoWindow;
  }
});
Object.defineProperty(exports, "AgmKmlLayer", {
  enumerable: true,
  get: function get() {
    return _kmlLayer.AgmKmlLayer;
  }
});
Object.defineProperty(exports, "AgmDataLayer", {
  enumerable: true,
  get: function get() {
    return _dataLayer.AgmDataLayer;
  }
});
Object.defineProperty(exports, "AgmMarker", {
  enumerable: true,
  get: function get() {
    return _marker.AgmMarker;
  }
});
Object.defineProperty(exports, "AgmPolygon", {
  enumerable: true,
  get: function get() {
    return _polygon.AgmPolygon;
  }
});
Object.defineProperty(exports, "AgmPolyline", {
  enumerable: true,
  get: function get() {
    return _polyline.AgmPolyline;
  }
});
Object.defineProperty(exports, "AgmPolylinePoint", {
  enumerable: true,
  get: function get() {
    return _polylinePoint.AgmPolylinePoint;
  }
});
Object.defineProperty(exports, "AgmFitBounds", {
  enumerable: true,
  get: function get() {
    return _fitBounds.AgmFitBounds;
  }
});

var _map = require("./directives/map");

var _circle = require("./directives/circle");

var _rectangle = require("./directives/rectangle");

var _infoWindow = require("./directives/info-window");

var _kmlLayer = require("./directives/kml-layer");

var _dataLayer = require("./directives/data-layer");

var _marker = require("./directives/marker");

var _polygon = require("./directives/polygon");

var _polyline = require("./directives/polyline");

var _polylinePoint = require("./directives/polyline-point");

var _fitBounds = require("./directives/fit-bounds");
